agent/app/service/container.go

```diff
@@ -1475,3 +1468,39 @@ func loadComposeCount(client *client.Client) int {
 
 	return len(composeMap)
 }
+func loadContainerPortForInfo(itemPorts []types.Port) []dto.PortHelper {
+	var exposedPorts []dto.PortHelper
+	samePortMap := make(map[string]dto.PortHelper)
+	ports := transPortToStr(itemPorts)
+	var itemPort dto.PortHelper
+	for _, item := range ports {
+		itemStr := strings.Split(item, "->")
+		if len(itemStr) < 2 {
+			continue
+		}
+		lastIndex := strings.LastIndex(itemStr[0], ":")
+		if lastIndex == -1 {
+			itemPort.HostPort = itemStr[0]
+		} else {
+			itemPort.HostIP = itemStr[0][0:lastIndex]
+			itemPort.HostPort = itemStr[0][lastIndex+1:]
+		}
+		itemContainer := strings.Split(itemStr[1], "/")
+		if len(itemContainer) != 2 {
+			continue
+		}
+		itemPort.ContainerPort = itemContainer[0]
+		itemPort.Protocol = itemContainer[1]
+		keyItem := fmt.Sprintf("%s->%s/%s", itemPort.HostPort, itemPort.ContainerPort, itemPort.Protocol)
+		if val, ok := samePortMap[keyItem]; ok {
+			val.HostIP = ""
+			samePortMap[keyItem] = val
+		} else {
+			samePortMap[keyItem] = itemPort
+		}
+	}
+	for _, val := range samePortMap {
+		exposedPorts = append(exposedPorts, val)
+	}
+	return exposedPorts
+}
```

Comment:
I'm having trouble analyzing the code provided since there is no actual source attached to it. Could you please share more details?